File: JebraAzureFunctions/JebraAzureFunctions/Models/UserSignInResponseModel.cs
using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace JebraAzureFunctions.Models
{
    class UserSignInResponseModel : Attribute
    {
        [JsonProperty]
        public int courseId { get; set; }
        [JsonProperty]
        public int userId { get; set; }
        [JsonProperty]
        public int instructorId { get; set; }
        [JsonProperty]
        public int stageId { get; set; }
    }
}

File: client-app/src/models/UserSignInResponseModel.ts
// Interface for UserSignIn JSON data returned from backend
export default interface UserSignInResponseModel {
    courseId: number,
    userId: number,
    instructorId: number,
    stageId: number
}

// Type guard for validating that data returned from the backend contains the expected fields
export function isUserSignInResponseModel(data: any): data is UserSignInResponseModel {
    if (typeof data !== "object" || Array.isArray(data) || data === null) {
        return false;
    }

    const usirData = data as UserSignInResponseModel;
    return typeof usirData.courseId === "number"
        && typeof usirData.userId === "number"
        && typeof usirData.instructorId == "number"
        && typeof usirData.stageId === "number";
}